ct/outline.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/burgerga/ProxmoxVE/add_outline/misc/build.func)
+# Copyright (c) 2024 community-scripts ORG
+# Author: Gerhard Burger (burgerga)
+# License: MIT
+# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+
+function header_info {
+clear
+cat <<"EOF"
+   ____        __  ___
+  / __ \__  __/ /_/ (_)___  ___
+ / / / / / / / __/ / / __ \/ _ \
+/ /_/ / /_/ / /_/ / / / / /  __/
+\____/\__,_/\__/_/_/_/ /_/\___/
+
+EOF
+}
+header_info
+echo -e "Loading..."
+APP="Outline"
+var_disk="8"
+var_cpu="2"
+var_ram="4096"
+var_os="debian"
+var_version="12"
+variables
+color
+catch_errors
+
+function default_settings() {
+  CT_TYPE="1"
+  PW=""
+  CT_ID=$NEXTID
+  HN=$NSAPP
+  DISK_SIZE="$var_disk"
+  CORE_COUNT="$var_cpu"
+  RAM_SIZE="$var_ram"
+  BRG="vmbr0"
+  NET="dhcp"
+  GATE=""
+  APT_CACHER=""
+  APT_CACHER_IP=""
+  DISABLEIP6="no"
+  MTU=""
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
+header_info
+check_container_storage
+check_container_resources
+if [[ ! -d /opt/outline ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+RELEASE=$(curl -s https://api.github.com/repos/outline/outline/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+if [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]] || [[ ! -f /opt/${APP}_version.txt ]]; then
+  msg_info "Stopping ${APP}"
+  systemctl stop outline
+  msg_ok "Stopped ${APP}"
+
+  msg_info "Updating ${APP} to ${RELEASE} (Patience)"
+  cd /opt
+  cp /opt/outline/.env /opt/.env
+  mv /opt/outline /opt/outline_bak
+  wget -q "https://github.com/outline/outline/archive/refs/tags/${RELEASE}.zip"
+  unzip -q ${RELEASE}.zip
+  mv outline-${RELEASE:1} /opt/outline
+  cd /opt/outline
+
+  ## Build in development mode
```

Comment:
Remove

Reply:
Happy to remove it, but I think it's a good idea in general to document your code. I tried to follow the [official instructions ](https://docs.getoutline.com/s/hosting/doc/from-source-BlBxrNzMIP), but in the end with some input from the Outline dev (https://github.com/outline/outline/discussions/8008#discussioncomment-11354747) and following the [linked Dockerfile](https://github.com/outline/outline/blob/main/Dockerfile.base) I was able to make it work. Is there a convention on where to document this so future maintainers are not  completely in the dark?

Reply:
I mean the comment ^^

Reply:
@havardthom see here for why I didn't follow official instructions

Reply:
Perhaps ask the outline maintainers if they could update their official docs on installing from source, then we wouldn't need to comment the code

Reply:
Yes, that would be ideal, I will reply to my original conversation with him. Apart from that I still think it would be good to think about how to include/link some form of development documentation. Knowing myself, six months from now I won't have a clue why I wrote things the way I did (and that's being generous :P)

Reply:
Fine for me. What do you say we leave the PR open until you have feedback?

Reply:
Works for me! In the mean time I will add the other requested changes, probably later this week.

Reply:
He updated the instructions, but only partially. Would still recommend the Dockerfile as reference.